src/librustc_mir/interpret/const_eval.rs

```diff
@@ -93,16 +93,23 @@ pub fn eval_body<'a, 'tcx>(
     }
 }
 
-pub fn value_to_const_value<'a, 'tcx>(
-    tcx: TyCtxt<'a, 'tcx, 'tcx>,
-    val: Value,
+pub fn value_to_const_value<'tcx>(
+    ecx: &EvalContext<'_, '_, 'tcx, CompileTimeEvaluator>,
+    mut val: Value,
     ty: Ty<'tcx>,
 ) -> &'tcx ty::Const<'tcx> {
-    let layout = tcx.layout_of(ty::ParamEnv::reveal_all().and(ty)).unwrap();
+    // Convert to ByVal or ByValPair if possible
```

Comment:
I've moved the code to convert to ByVal and ByValPair in here.

Reply:
so https://github.com/rust-lang/rust/pull/50710/files#diff-2a7855a4ad9a8ae6de5b6c25045a41d7R213 is obsolete now?

Reply:
Possibly. It is obsolete when it's called from `const_eval` at least.